Dockerfile

```diff
@@ -0,0 +1,41 @@
+# Multipart build dockerfile to build and serve datagateway
+
+FROM node:16-alpine as build
+WORKDIR /datagateway
+ENV PATH /datagateway/node_modules/.bin:$PATH
+
+# TODO: use yarn install --production:
+# https://github.com/ral-facilities/datagateway/issues/1155
+
+# Install dependancies
+COPY . .
+RUN yarn install
+RUN yarn build
+
+# Put the output of the build into an apache server
+FROM httpd:alpine
```

Comment:
Should we specify a version here? I am not sure if it just grabs the latest version if a version isn't specified but if it does then that may potentially cause issues when new versions are released in future.

Reply:
I tested Node 17 but the build failed due to some of the build tools not yet supporting it. Sticking to Node 16 for now, in fact the [release build workflow](https://github.com/ral-facilities/datagateway/blob/master/.github/workflows/release-build.yml) uses Node 14 so it doesn't seem to matter too much.

I have configured for Yarn 1 (version 1.22.18) to be used for now and added a TODO to upgrade to Yarn 2 once the project is upgraded to Yarn 2 which will happen after [React 17 and MUIv5 upgrade #830 #1096](https://github.com/ral-facilities/datagateway/pull/1129) is merged. I will add a note to the wiki.

I also set a version for `httpd`.